src/stan_math_signatures/Generate.ml

```diff
@@ -2703,3 +2550,41 @@ let () =
     ~required_fn_rt:UnsizedType.UVector
     ~required_fn_args:[UnsizedType.(AutoDiffable, UVector)]
     ()
+
+(** Print a module definition to [file]
+  that contains the signatures computed above *)
+let generate_module file =
+  let marshal e = Marshal.to_string e [] in
+  (* Core's Hashtbl cannot be safely Marshal'd, so we
+     round trip through an associative list *)
```

Comment:
From the OCaml forum discussion it's unclear if `Stdlib.Hashtbl` can be Marshal'ed. Did that fail? Or is Stdlib undesirable for some other reason? (it doesn't seem to have `Hashtbl.find_multi`...)
What about converting to `Core.String.Map`?

Reply:
Core's `Map` also has functional values that can't be serialized, unfortunately. 

I've tried both Map and Hashtbl from stdlib, they both perform basically equally to the current code, but require some more ugly contortions to use (because of the absence of `find_multi`, but also the absense of things like `.keys` or `.to_list`...)

The interesting thing this also made me realize is that we almost never do direct lookups into this table. The Typechecker entirely uses `Environment.ml`, which converts it back into an alist first. Especially if we were willing to sort during generation, I suspect we could avoid creating a map/hashtable entirely. But I think that might be better off waiting for another time

Reply:
I forgot that cons lists don't have constant time access, so you can't just binary search them. Ignore the second half of the above comment